packages/medusa-payment-adyen/src/api/routes/store/authorize-payment.js

```diff
@@ -1,10 +1,15 @@
 import { Validator, MedusaError } from "medusa-core-utils"
```

Comment:
I think we should move this endpoint to the core. 

The only parts that are payment provider specific in this controller is:

```js
     // Shopper IP address for risk valuation
     const shopperIp = requestIp.getClientIp(req)
```

and 

```js
    // MongoDB does not allow us to store keys with dots
   if (authorizedPayment.additionalData) {
       delete data.additionalData["recurring.shopperReference"]
       delete data.additionalData["recurring.recurringDetailReference"]
   }
```

We should be able to move this into the `authorizePayment` method defined in the payment provider services.

Reply:
We might have to add a `context` argument to `authorizePayment` so that we can pass details about the request. In this case for example details about the IP address.